### PR TITLE
Add days in year at loan level #FINERACT-1302

### PR DESCRIPTION
**Pull request by davidyaha**
## Description
FINERACT-1302 Added days in year at loan level.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by davidyaha** [comment]
@vorburger @awasum  would love your review on this one. thanks!

**Comment by vorburger** [comment]
@avikganguly01 saw that you were active in #1586 (welcome "back"), would you be able to review this PR from a promising new contributor, to support the ecosystem? Thank You!

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java:492]: O don't know the details of this code, so this a general review which someone else with more in-depth knowledge may want to comment on, but years normally have 365 or 366 (leap year) days.. what's 364 used for, and 360 - and 1?!

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java:339]: Nit pick: remove the = null, it's not needed (because it gets assigned in either case), nicer to read, AND safer (if the code below EVER changes, it would cause a compilation failure instead of potential NPE). Makes sense?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoansApiResourceSwagger.java
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoansApiResourceSwagger.java:518]: As below, curious values? Perhaps for a good reason which I don't understand.

**Review by vorburger — CHANGES_REQUESTED** [review]
Some feedback. Also, would an IT for this be possible? Just asking!

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java
Comment (ptuomola) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java:492]: Hi @vorburger - that's the joy of archaic day count conventions for interest rate calculations :-). There are hundreds of conventions but 365 and 360 are the most common. See https://www.investopedia.com/terms/d/daycount.asp (or http://strata.opengamma.io/apidocs/com/opengamma/strata/basics/date/DayCounts.html for some more exotic ones :-)).

Comment (vorburger) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java:492]: @ptuomola oh what fun. OK, then I'll Resolve this.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java
Comment (vorburger) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java:339]: @davidyaha it looks like you haven't had time to address this feedback. As it's only minor, I'll not let this hold up merging.

**Comment by vorburger** [comment]
@davidyaha I don't know enough about the full details of this code, and unfortunately neither @avikganguly01 nor any other remaining active maintainer has reviewed this in the last 2 weeks. You look like you know what you are doing, and we want to encourage new contributions, so I'll go ahead and merge this now.

PS: Adding tests when you can would be great (in future contributions).

**Comment by edcable** [comment]
@vorburger @Nayan and members of the conflux team are becoming active members of the community again so I'll try to have them do a review of this commit as well.

**Pull request merged**
